documentation/modules/snippet_secret-cr.adoc

```diff
@@ -0,0 +1,65 @@
+:_content-type: SNIPPET
+
+
+[source,yaml,subs="attributes+"]
+----
+$ cat << EOF | {oc} apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <secret>
+  namespace: <namespace>
+  ownerReferences: <1>
+    - apiVersion: forklift.konveyor.io/v1beta1
+      kind: Provider
+      name: <provider_name>
+      uid: <provider_uid>
+  labels:
+ifdef::vmware
+    createdForProviderType: vsphere
+endif::
+ifdef::rhv
+    createdForProviderType: ovirt
+endif::
+    createdForResourceType: providers
+type: Opaque
+stringData: <3>
+  user: <user> <4>
+  password: <password> <5>
+  insecureSkipVerify: <true/false> <6>
+  domainName: <domain_name> <7>
+  projectName: <project_name> <8>
+  regionName: <region name> <9>
+  cacert: | <10>
+    <ca_certificate>
+  url: <api_end_point> <11>
+  thumbprint: <vcenter_fingerprint> <12>
+  token: <service_account_bearer_token> <13>
+EOF
+----
+<1> The `ownerReferences` section is optional.
+<2> Specify the type of source provider. Allowed values are `ovirt`, `vsphere`, `openstack`, `ova`, and `openshift`. This label is needed to verify the credentials are correct when the remote system is accessible and, for {rhv-short}, to retrieve the {manager} CA certificate when a third-party certificate is specified.
+<3> The `stringData` section for OVA is different and is described in a note that follows the description of the `Secret` manifest.
+<4> Specify the vCenter user, the {rhv-short} {manager} user, or the {osp} user.
+<5> Specify the user password.
+<6> Specify `<true>` to skip certificate verification, which proceeds with an insecure migration and then the certificate is not required. Insecure migration means that the transferred data is sent over an insecure connection and potentially sensitive data could be exposed. Specifying `<false>` verifies the certificate.
+<7> {osp} only: Specify the domain name.
+<8> {osp} only: Specify the project name.
+<9> {osp} only: Specify the name of the {osp} region.
+<10> {rhv-short} and {osp} only: For {rhv-short}, enter the {manager} CA certificate unless it was replaced by a third-party certificate, in which case enter the {manager} Apache CA certificate. You can retrieve the {manager} CA certificate at https://<engine_host>/ovirt-engine/services/pki-resource?resource=ca-certificate&format=X509-PEM-CA. For {osp}, enter the CA certificate for connecting to the source environment. The certificate is not used when `insecureSkipVerify` is set to `<true>`.
```

Comment:
Perhaps it is worth mentioning that when `insecureSkipVerify` is false and cacert is not set we try to use the system CA

Reply:
@bennyz  Thanks. In the end, I am not using this snippet. I will mention that in the OpenStack section in the next iteration.

Reply:
@RichardHoch please note that this part is not specific to OpenStack

Reply:
@ahadas I am not using this this snippet -- please review the previews or the file new-migrating-virtual-machines-cli.adoc  .